internal/transport/http2_server.go

```diff
@@ -630,18 +630,24 @@ func (t *http2Server) HandleStreams(handle func(*Stream), traceCtx func(context.
 				continue
 			}
 			if err == io.EOF || err == io.ErrUnexpectedEOF {
+				if logger.V(logLevel) {
+					logger.Infof("transport: closing the server transport due to connection level EOF.")
+				}
```

Comment:
Same comments as in the client, maybe.  I'm not sure if passing an error to `t.Close` is better, purely for logging purposes, and to keep the code consistent, but it might be.

Reply:
Actually I switched Server Side Transports to take error on Close and log it. Thanks for the suggestion.